src/db/collections/outputs/mod.rs

```diff
@@ -329,3 +332,285 @@ impl MongoDb {
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "$nor": [
+                            { "metadata.booked.milestone_timestamp": { "$lt": start_timestamp } },
+                            { "metadata.booked.milestone_timestamp": { "$gte": end_timestamp } },
+                        ],
+                        "output.kind": "nft"
+                    } },
+                    // First group the nfts by their ids
+                    doc! { "$group" : {
+                        "_id": "$output.nft_id",
+                        "amount": { "$sum": { "$toDouble": "$output.amount" } },
+                    }},
+                    doc! { "$group" : {
+                        "_id": null,
+                        "count": { "$sum": 1 },
+                        "total_value": { "$sum": "$amount" },
+                        "avg_value": { "$avg": "$amount" },
+                    }},
+                ],
+                None,
+            )
+            .await?
+            .try_next()
+            .await?
+            .map(bson::from_document)
+            .transpose()?
+            .unwrap_or_default())
+    }
+
+    /// Gathers nft output analytics.
+    pub async fn get_foundry_analytics(
+        &self,
+        start_timestamp: Option<MilestoneTimestamp>,
+        end_timestamp: Option<MilestoneTimestamp>,
+    ) -> Result<OutputAnalyticsResult, Error> {
+        Ok(self
+            .0
+            .collection::<OutputAnalyticsResult>(OutputDocument::COLLECTION)
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "$nor": [
+                            { "metadata.booked.milestone_timestamp": { "$lt": start_timestamp } },
+                            { "metadata.booked.milestone_timestamp": { "$gte": end_timestamp } },
+                        ],
+                        "output.kind": "foundry"
+                    } },
+                    doc! { "$unwind": "$output.native_tokens" },
+                    // First group by token id
+                    doc! { "$group" : {
+                        "_id": "$output.native_tokens.token_id",
+                        "amount": { "$sum": "$output.native_tokens.float_amount" },
+                    }},
+                    doc! { "$group" : {
+                        "_id": null,
+                        "count": { "$sum": 1 },
+                        "total_value": { "$sum": "$amount" },
+                        "avg_value": { "$avg": "$amount" },
+                    }},
+                ],
+                None,
+            )
+            .await?
+            .try_next()
+            .await?
+            .map(bson::from_document)
+            .transpose()?
+            .unwrap_or_default())
+    }
+
+    /// Gathers locked storage deposit analytics.
+    pub async fn get_locked_storage_deposit_analytics(
```

Comment:
We should only compute that for the current milestone. Also, we need to sum the `OutputDetails::num_key_bytes` and `OutputDetails::num_data_bytes` of each output. Once we have those fields we can compute the byte cost by looking at the parameters in `StatusDocument`.

Reply:
What do you mean by only computing it for the current milestone?